### ISSUE TITLE
Request to include new 32 sample disk filtering in shaders fix compilation

### ISSUE THREAD

**Issue by Mikan1520** [RaphaelK12/GTA4-Improved-Soft-Shadows#1]
Hi, I know this isn't an "issue", and the license permits it, but I'd rather ask as a courtesy. A friend and I are working on a compilation of shader fixes for the game, and we were wondering if, among our lighting shaders, we can include your 32 sample disk shadows (or any of them, whichever we feel looks best)

**Comment by RaphaelK12** [comment]
Hi, I'm very happy to be able to contribute in some way to improving this amazing game.

If you have any questions just ask and I'll answer whenever possible, I also accept suggestions for improvements.

When possible send me the link of your shader compilation, I'm curious.

What did you think of the result?
Did the shaders work well for you?

**Comment by Mikan1520** [comment]
Yeah they all look really nice! I will send a link when we upload to gtaforums! Thanks so much!

**Comment by Mikan1520** [comment]
I am curious how much they could be softened, if you're able to answer that question. The shadows on older patches were quite soft, and if possible I'd be interested in matching that level of softness

**Comment by RaphaelK12** [comment]
I only know versions 1.0.8.0 and 1.2.0.43 that have horrible shadows, which motivated me to create these shaders.
Which version has nice soft shadows?

At the moment I have 2 good alternatives:
"New 32 sample disk" which is an improvement over the original shadows using more samples but doesn't change the sampling type and you can still see the square pixels.

"Very Soft Shadow ( 5x5 Gausian Blur)" makes the shadows very soft using a bilinear filter on a 5x5 pixel grid, it still has artifacts in some regions, but it's the best result I've achieved so far.

**Comment by RaphaelK12** [comment]
I researched a lot of GTA4 shadows from different versions, but I didn't find any that had soft shadows, the closest was dithered shadows from version 1.0.1 to 1.0.4, which had a softer look.
From version 1.0.7 it had a higher sharpness but the shadow system was modified by removing the dithering, adding a "poisson disk sampling" of 12 samples and making the shadow pixels visibly square.

Version comparison:
https://www.youtube.com/watch?v=aqa9qC-Iw4A&t=102s

In GTA Garage I found a mod for version 1.0.1 to 1.0.4 that uses the native Dithering of the shadows but increasing the amount of samples, making the shadows have a good quality in my point of view.
https://www.gtagarage.com/mods/show.php?id=5691

**Comment by Mikan1520** [comment]
1.0.4.0 softness is what I was thinking of, the amount of visual noise in it reduced the shadow penumbra sharpness to a degree that I feel fits the game's atmosphere really well, and it would be interesting to see a softness similar to that without directly copying it

**Comment by RaphaelK12** [comment]
I'm trying to get any version from 1.0.4.0 or older to work, but none are working.

I noticed that several modifications were made from version 1.0.4 to 1.0.7, not only in the shaders, because the shadow projection mode also changed, making the shadow move according to the direction in which you look, with the objective of having a higher resolution close to the point of view, but making the edges of shadows jitter as you move, something that in version 1.0.4 appeared to be stable and brought less resolution close to the player but had a constant resolution with no clipping between near and far shadows.

I'm testing using 16-sample dithering which might give you the result you expect, but I have no idea how to improve the big difference part between near and far shadows, so expect some glitches.

**Comment by RaphaelK12** [comment]
Hi
Two new shadows available.

https://github.com/RaphaelK12/GTA4-Improved-Soft-Shadows/tree/main/Shaders/Dot%20Dither%2016%20samples%20(noise)
Something similar to version 1.0.3, but without the dithering patterns, just random noise.
It's not quite what I wanted, the shadows aren't stable at the moment, but it's working and producing a result close to what I expected, I'm getting close.

Low
![image](https://github.com/RaphaelK12/GTA4-Improved-Soft-Shadows/assets/38438130/4a3e46c4-49e9-40d8-9dc9-99cbabd7b7cf)

Medium
![image](https://github.com/RaphaelK12/GTA4-Improved-Soft-Shadows/assets/38438130/fbae713a-2ad6-4476-8d19-6c05a973af10)


Am I going the right way?

**Comment by Mikan1520** [comment]
Woah very nice, yeah that's looking great! Thanks for giving it a go, I really appreciate it lol

**Comment by Parallellines0451** [comment]
Looks great. Excellent work Raphael, your shaders are a life saver for sure.

By the way, are you on Discord by any chance?

**Comment by RaphaelK12** [comment]
> Woah very nice, yeah that's looking great! Thanks for giving it a go, I really appreciate it lol

Thank you, any suggestions will be appreciated.



> Looks great. Excellent work Raphael, your shaders are a life saver for sure.

Thanks



> By the way, are you on Discord by any chance?

I'm rarely on discord, I have little time available and when I'm on discord it's usually playing online with friends.
I'm Brazilian and I speak Brazilian Portuguese.
You understand Portuguese? I don't speak English and I understand little, I just read a lot in English and I know how to write a little (that's why I write a lot of things wrong lol).

**Comment by RaphaelK12** [comment]
> By the way, are you on Discord by any chance?

Rambo#3149

**Comment by Parallellines0451** [comment]
I'm brazilian too. Figured you'd speak portuguese as well judging by your remarks in the shaders XD

Your work so far has been amazing and I'm eager to see what else you'll cook up in the future.
